### PR TITLE
Series/limits : Added heuristics to deal with limit computation of undefined functions

### LINKED ISSUE #7391: Limits for expressions with undetermined functions give wrong results
I'm trying to compute limits of functions of several variables. These are limits when the variables converge to a particular point p, and in general they exist only when the value is the same on all possible paths towards p.

Sympy computes limits only for functions of a single free variable, so I express all variables in the original function as functions of a free variable. An example:

``` Python
func = x*y*y / (x*x + y**4)
func2 = func.subs(y, f(x))
limit(func2, x, 0)
=> 0
```

This gives me zero as result. However, zero is not the right result for _all_ functions `f`. In particular:

```
func3 = func.subs(y, sqrt(x))
limit(func3, x, 0)
=> 1/2
```

This is a special case, and for other choices of `f` the limit _is_ zero... still it would be nice if sympy gave me a result that is true in all cases.

### LINKED ISSUE #8166: Limit assumes function is continuous?
The following has surprised me:

``` python
x = Symbol('x')
f = Function('f')
print limit(f(x), x, 4)  ## prints f(4)
```

Is that the expected behaviour?

I think that [Mathematica](http://reference.wolfram.com/language/ref/Limit.html) does not assume a generic function is continuous, and the result that Wolfram Alpha gives is

```
(no result found in terms of standard mathematical functions)
```

I believe that a better result would be simply to keep the limit unevaluated, unless we know something about `f`.

### PR DESCRIPTION
**Pull request by anutosh491**
#### References to other Issues or PRs

Closes #8166
Closes #7391

#### Brief description of what is fixed or changed
On Master
```
>>> x = Symbol('x')
... f = Function('f')
... print limit(f(x), x, 4)  
f(4)

>>> func = x*y*y / (x*x + y**4)
... func2 = func.subs(y, f(x)) 
... limit(func2, x, 0) # won't work generally eg f(x) = sqrt(x)
0
```
Obviously these results are not what one would expect as we don't make any assumptions about the function and its continuity, hence there should be heuristics for the same !

On Committed branch
```
>>> x = Symbol('x')
>>> f = Function('f')
>>> limit(f(x), x, 4)
Traceback (most recent call last):
NotImplementedError: Result depends on the continuity of f(x) at point 4
>>> 
>>> y = Symbol('y')
>>> func = x*y*y / (x*x + y**4)
>>> func2 = func.subs(y, f(x))
>>>
>>> limit(func2, x, 0)
Traceback (most recent call last):
NotImplementedError: Result depends on the continuity of x*f(x)**2/(x**2 + f(x)**4) at point 0
>>>
>>> func3 = func.subs(y, sqrt(x))
>>> limit(func3, x, 0)
1/2
```
#### Release Notes

* series
  * Added heuristics to deal with limit computation of undefined functions.

### CODE REVIEW

**Comment by anutosh491** [comment]
ping @smichr @jksuom @oscarbenjamin !

**Comment by ThePauliPrinciple** [comment]
How would you now indicate that your function is continuous? I think that there are many cases where you would want to get `limit(f(x), x, 4) == f(4)`

File: .mailmap [hunk 1]
```diff
@@ -1233,7 +1233,7 @@ alijosephine <alijosephine@gmail.com>
 amsuhane <ayushsuhane99@iitkgp.ac.in> amsuhane <“ayushsuhane99@iitkgp.ac.in”>
 anca-mc <anca-mc@users.noreply.github.com>
 andreo <andrey.torba@gmail.com>
-anutosh491 <andersonbhat491@gmail.com>
+anutosh491 <andersonbhat491@gmail.com> Anutosh Bhat <87052487+anutosh491@users.noreply.github.com>
```
Comment (czgdp1807) [inline .mailmap:1236]:
```suggestion
Anutosh Bhat <andersonbhat491@gmail.com>
```

**Comment by anutosh491** [comment]
cc @arnabnandikgp would you like to finish this (maybe open a new PR for the same). Don't exactly remember the issues here but you should be able to address this !

**Pull request closed without merging**